File: Difficulty: Easy/Non Repeating Character/non-repeating-character.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;

class Driverclass {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            String st = sc.next();

            char ans = new Solution().nonrepeatingCharacter(st);

            if (ans != '$')
                System.out.println(ans);
            else
                System.out.println(-1);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    // Function to find the first non-repeating character in a string.
    static char nonrepeatingCharacter(String s) {
        // Your code here
        char[] c=s.toCharArray();
        for(int i=0;i<c.length;i++){
            int count=0;
            for(int j=0;j<c.length;j++){
                if(c[i]==c[j]){
                    count++;
                }
            }
            if(count==1){
                return c[i];
            }
        }
        return '$';
    }
}

File: Difficulty: Easy/Non Repeating Character/non-repeating-character.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    // Function to find the first non-repeating character in a string.
    char nonRepeatingChar(string &s) {
        
        vector<int>count(256,0);
        for(int i=0;i<s.size();i++){
            count[s[i]]++;
        }
        for(int i=0;i<s.size();i++){
            if(count[s[i]]==1){
                return s[i];
            }
        }
        return '$';
        
    }
};

    //   //Your code here
    //   vector<int>count(256,0);
    //   for(int i=0;i<S.size();i++){
    //       count[S[i]]++;
    //   }
    //   for(int i=0;i<S.size();i++){
    //       if(count[S[i]]==1){
    //           return S[i];
    //       }
    //   }
    //   return '$';
    // }


//{ Driver Code Starts.

int main() {

    int T;
    cin >> T;

    while (T--) {

        string S;
        cin >> S;
        Solution obj;
        char ans = obj.nonRepeatingChar(S);

        if (ans != '$')
            cout << ans;
        else
            cout << "-1";

        cout << endl;

        cout << "~"
             << "\n";
    }

    return 0;
}

// } Driver Code Ends